### PR TITLE
when running conformance tests log envoy config changes

### PR DESCRIPTION
**Pull request by dprotaso**
- we do this via a sidecar on the envoy daemonset and poll
  the admin endpoint. output goes to stdout

### CODE REVIEW

File: hack/update-deps.sh [hunk 1]
```diff
@@ -111,6 +111,7 @@ git apply ${ROOT_DIR}/hack/contour.patch
 # We do this manually because it's challenging to rewrite
 # the ClusterRoleBinding without collateral damage.
 cat > config/contour/internal.yaml <<EOF
+---
```
Comment (dprotaso) [inline hack/update-deps.sh:114]: this lets us `cat *.yaml` the files together

File: test/cmd/config-dumper/main.go [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+	config, err := normalizeJSON(res.Body)
+	if err != nil {
+		log.Printf("error reading & normalizing json: %s", err)
+		return
+	}
+
+	if currentConfig != config {
+		log.Printf("updated config: %s", config)
+		currentConfig = config
+		lastUpdate = time.Now()
```
Comment (mattmoor) [inline test/cmd/config-dumper/main.go:84]: I'd be curious how good `kmp.Diff` is at showing deltas here.

Comment (dprotaso) [reply test/cmd/config-dumper/main.go:84]: when diffing the normalized it looked like a hex editor output - I'll see how the diff is when it's not normalized that may be more friendly

**Comment by dprotaso** [comment]
ok done
```
2020/08/12 10:34:57 diff:   (
  	"""
  	... // 839 identical lines
  	      "cluster": {
  	       "@type": "type.googleapis.com/envoy.api.v2.Cluster",
- 	       "alt_stat_name": "serving-tests_ingress-conformance-10-retry-period-5-attempts-5-dmxnrheb_85",
+ 	       "alt_stat_name": "serving-tests_ingress-conformance-10-retry-period-3-attempts-2-zhpzmiqn_64",
  	       "common_lb_config": {
  	        "healthy_panic_threshold": {}
  	... // 13 identical lines
  	         }
  	        },
- 	        "service_name": "serving-tests/ingress-conformance-10-retry-period-5-attempts-5-dmxnrheb/http"
+ 	        "service_name": "serving-tests/ingress-conformance-10-retry-period-3-attempts-2-zhpzmiqn/http"
  	       },
- 	       "name": "serving-tests/ingress--372e83/85/da39a3ee5e",
+ 	       "name": "serving-tests/ingress--63a19e/64/da39a3ee5e",
  	       "type": "EDS"
  	      },
- 	      "last_updated": "2020-08-12T14:34:52.946Z",
- 	      "version_info": "1214"
+ 	      "last_updated": "2020-08-12T14:34:57.644Z",
+ 	      "version_info": "1226"
  	     },
  	     {
  	      "cluster": {
  	       "@type": "type.googleapis.com/envoy.api.v2.Cluster",
- 	       "alt_stat_name": "serving-tests_ingress-conformance-10-retry-period-3-attempts-2-zhpzmiqn_64",
+ 	       "alt_stat_name": "serving-tests_ingress-conformance-7-retry-period-10-attempts-9-dcfdclfq_97",
  	       "common_lb_config": {
  	        "healthy_panic_threshold": {}
  	... // 13 identical lines
  	         }
  	        },
- 	        "service_name": "serving-tests/ingress-conformance-10-retry-period-3-attempts-2-zhpzmiqn/http"
+ 	        "service_name": "serving-tests/ingress-conformance-7-retry-period-10-attempts-9-dcfdclfq/http"
  	       },
- 	       "name": "serving-tests/ingress--63a19e/64/da39a3ee5e",
+ 	       "name": "serving-tests/ingress--676945/97/da39a3ee5e",
  	       "type": "EDS"
  	      },
- 	      "last_updated": "2020-08-12T14:34:57.644Z",
- 	      "version_info": "1226"
+ 	      "last_updated": "2020-08-12T14:29:59.710Z",
+ 	      "version_info": "934"
  	     },
  	     {
  	      "cluster": {
  	       "@type": "type.googleapis.com/envoy.api.v2.Cluster",
- 	       "alt_stat_name": "serving-tests_ingress-conformance-7-retry-period-10-attempts-9-dcfdclfq_97",
+ 	       "alt_stat_name": "serving-tests_ingress-conformance-10-retry-period-6-attempts-6-yecalrrw_63",
  	       "common_lb_config": {
  	        "healthy_panic_threshold": {}
  	... // 13 identical lines
```

**Comment by dprotaso** [comment]
@mattmoor I made this PR to help you debug - do you still need/want these changes?

**Comment by mattmoor** [comment]
I'd say let's keep them in our back pocket if we run afoul of 503s again, but we're looking super green lately.

**Comment by dprotaso** [comment]
cool

**Pull request closed without merging**